extensions-core/stats/src/main/java/io/druid/query/aggregation/teststats/ZtestPostAggregator.java

```diff
@@ -0,0 +1,159 @@
+      @JsonProperty("fields") List<PostAggregator> fields
+  ) {
+    Preconditions.checkNotNull(name, "Must have a valid, non-null post-aggregator name");
+    Preconditions.checkArgument(fields.size() == 4, "Must have 4 fields; " +
+        "\"fields\": <total success count of sample 1>, <sample size of population 1>," +
+        " <total success counts of sample 2>, <sample size of population 2>");
+    this.name = name;
+    this.fields = fields;
+  }
+
+  @Override
+  public Set<String> getDependentFields() {
+    Set<String> dependentFields = Sets.newLinkedHashSet();
+    for (PostAggregator field : fields) {
+      dependentFields.addAll(field.getDependentFields());
+    }
+    return dependentFields;
+  }
+
+  @Override
+  public Comparator getComparator() {
+    return ArithmeticPostAggregator.DEFAULT_COMPARATOR;
+  }
+
+  @Override
+  public Object compute(Map<String, Object> combinedAggregators) {
+
+    PostAggregator[] args = new PostAggregator[fields.size()];
+    for (int j = 0; j < fields.size(); j++) {
+      args[j] = fields.get(j);
+    }
+
+    return zScoreTwoSamples(
+        ((Number) args[0].compute(combinedAggregators)).doubleValue(),
+        ((Number) args[1].compute(combinedAggregators)).doubleValue(),
+        ((Number) args[2].compute(combinedAggregators)).doubleValue(),
+        ((Number) args[3].compute(combinedAggregators)).doubleValue());
+  }
+
+  @Override
+  @JsonProperty
+  public String getName() {
+    return name;
+  }
+
+  @Override
+  public PostAggregator decorate(Map<String, AggregatorFactory> aggregators) {
+    return this;
+  }
+
+  @JsonProperty
+  public List<PostAggregator> getFields() {
+    return fields;
+  }
+
+  /**
+   * 1. calculating zscore for two-sample Z test. IOW,
+   *    using z-test statistic for testing the difference of two population proportions.
+   * 2. converting binary variables (e.g. success or not) to continuous variables (e.g. conversion rate).
+   *
+   * @param the success count of population 1
+   * @param param sample size of population 1
+   * @param the success count of population 2
+   * @param sample size of population 2
+   */
+  private double zScoreTwoSamples(Double s1count, Double p1count, Double s2count, Double p2count) {
+    double convertRate1;
+    double convertRate2;
+    Preconditions.checkState(s1count >= 0, "success count can't be negative.");
+    Preconditions.checkState(s2count >= 0, "success count can't be negative.");
+    Preconditions.checkState(p1count >= s1count, "sample size can't be smaller than the success count.");
+    Preconditions.checkState(p2count >= s2count, "sample size can't be smaller than the success count.");
+
+    try {
+      convertRate1 = s1count / p1count;
+      convertRate2 = s2count / p2count;
+
+      return (convertRate1 - convertRate2) /
+          Math.sqrt((convertRate1 * (1 - convertRate1) / p1count) +
+              (convertRate2 * (1 - convertRate2) / p2count));
+    } catch (Exception ex) {
```

Comment:
What kind of Exception can be thrown here? Could this be made more specific?

Also, is it possible to use some other mechanism of detecting whatever problem this exception is detecting? Exceptions are quite slow compared to other forms of flow control and so I'm concerned about using them in a post-aggregator `compute` function.